packages/web-components/fast-foundation/src/date-picker/date-picker.ts

```diff
@@ -0,0 +1,1261 @@
+    DOM,
+    observable,
+    SyntheticViewTemplate,
+} from "@microsoft/fast-element";
+import {
+    keyArrowDown,
+    keyArrowLeft,
+    keyArrowRight,
+    keyArrowUp,
+    keyEnter,
+    keyEscape,
+    keySpace,
+} from "@microsoft/fast-web-utilities";
+import type {
+    FoundationElementDefinition,
+    FoundationElementTemplate,
+} from "../foundation-element";
+import type { ListboxElement } from "../listbox";
+import {
+    DateFormatter,
+    DayFormat,
+    MonthFormat,
+    WeekdayFormat,
+    YearFormat,
+} from "../calendar/date-formatter";
+import type { TextField } from "../text-field";
+import { FormAssociatedDatePicker } from "./date-picker.form-associated";
+
+/**
+ * Date picker options
+ * @alpha
+ */
+export type DatePickerOptions = FoundationElementDefinition & {
+    calendarIcon?:
+        | FoundationElementTemplate<
+              SyntheticViewTemplate<any, DatePicker>,
+              DatePickerOptions
+          >
+        | SyntheticViewTemplate
+        | string;
+    previousIcon?:
+        | FoundationElementTemplate<
+              SyntheticViewTemplate<any, DatePicker>,
+              DatePickerOptions
+          >
+        | SyntheticViewTemplate
+        | string;
+    nextIcon?:
+        | FoundationElementTemplate<
+              SyntheticViewTemplate<any, DatePicker>,
+              DatePickerOptions
+          >
+        | SyntheticViewTemplate
+        | string;
+};
+
+/**
+ * Struct for holding parsed data from a specific Date
+ * @public
+ */
+export type DateData = {
+    year?: number;
+    month?: number;
+    day?: number;
+    hour?: number;
+    minute?: number;
+    meridian?: "AM" | "PM";
+};
+
+/**
+ * A Date picker component
+ * @alpha
+ */
+export class DatePicker extends FormAssociatedDatePicker {
+    /**
+     * Date picker type
+     * @public
+     */
+    @attr
+    public type: "date" | "datetime-local" | "month" | "year" | "time" = "date";
+
+    /**
+     * Ensures that the value is valid for the updated type
+     * @param previous - previous type
+     * @param next - updated type
+     * @public
+     */
+    public typeChanged(previous: string, next: string): void {
+        if (next && this.value) {
+            this.valueChanged("", this.value);
+        }
+    }
+
+    /**
+     * Value for the date-picker
+     * @public
+     */
+    @attr
+    public value: string;
+
+    /**
+     * Validates that the value is a valid date
+     * @param previous - the previous selected value
+     * @param next - the updated selected value
+     * @publc
+     */
+    public valueChanged(previous: string, next: string): void {
+        super.valueChanged(previous, next);
+        if (previous !== next) {
+            if (!previous) {
+                this.setValue(next);
+            }
+        }
+    }
+
+    /**
+     * Text used for resetting the views back to todays date
+     * @public
+     */
+    @attr({ attribute: "reset-text" })
+    public resetText: string = "Go to today";
```

Comment:
We should not be hard-coding English strings

Reply:
I wanted to have at least something by default. I've replaced it with the current date as the default, formatted using the locale set up.